xarray/backends/h5netcdf_.py

```diff
@@ -69,8 +69,40 @@ class H5NetCDFStore(WritableCFDataStore):
     """Store for reading and writing data via h5netcdf
     """
 
-    def __init__(
-        self,
+    __slots__ = (
+        "autoclose",
+        "format",
+        "is_remote",
+        "lock",
+        "_filename",
+        "_group",
+        "_manager",
+        "_mode",
+    )
+
+    def __init__(self, manager, group=None, mode=None, lock=HDF5_LOCK, autoclose=False):
+
+        import h5netcdf
+
+        if isinstance(manager, (h5netcdf.File, h5netcdf.Group)):
+            if group is None:
+                root, group = manager._root, manager.name
```

Comment:
I think it's probably a better idea to use the `find_root_and_group` helper function that we use for netCDF4. Accessing private attributes like `_root` is a little unsafe.

Reply:
#3680 needs to be resolved first. I appreciated any guidance on that blocking issue.